### DISCUSSION TITLE
Native vim.snippet integration?

### DISCUSSION THREAD

**Discussion by Anrock** [hrsh7th/nvim-cmp#1768] [Ideas]
Never used snippets and nvim-cmp forcing me to install, configure and deal with additional plugin kinda bugs me.

However native snippet support recently landed in HEAD and with
```
cmp.setup({
  snippet = {
    expand = function(args)
      -- Native neovim snippet support
      vim.snippet.expand(args.body)
    end
  },
```
It kinda works (except I don't know how to use snippets so there are snippet leftovers everywhere lol).

AFAIK native snippet support intended to be minimal barebones for LSP which aligns with nvim-cmp explanation why snippet plugin is mandatory - for LSP to work.

I guess nvim-cmp can use native snippet when they're released as default "I don't want any extras" option? Just needs some documentation update.

**Comment by wookayin** [comment]
Yes, from neovim 0.10 this will work well and should be the default option (no dependencies on external snippet plugins). It's yet a nightly feature.
